### PR TITLE
make single script for PowerNVEEH.py and PowerVMEEH.py

### PR DESCRIPTION
**Pull request by smruti77**
added EEH.py test by combined both PowerNVEEH.py and PowerVMEEH.py
and removed both the test

Signed-off-by: bismurti bidhibrata pattajoshi <bbidhibr@in.ibm.com>

### CODE REVIEW

**Comment by smruti77** [comment]
[EEH_powerNV.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5734853/EEH_powerNV.log)
[EEH_powerVM.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/5734854/EEH_powerVM.log)

File: io/pci/EEH.py [hunk 1]
```diff
@@ -25,27 +25,29 @@
...
-    def __init__(self, msg, dev, log=None):
-        self.msg = msg
+    def __init__(self, thing, dev, log=None):
+        self.thing = thing
         self.dev = dev
         self.log = log
 
     def __str__(self):
-        return "%s %s recovery failed: %s" % (self.msg, self.dev, self.log)
+        return "%s %s recovery failed: %s" % (self.thing, self.dev, self.log)
```
Comment (narasimhan-v) [inline io/pci/EEH.py:47]: A better name instead of `thing` please

File: io/pci/EEH.py [hunk 2]
```diff
@@ -220,10 +244,20 @@ def check_eeh_removed(cls):
         """
         tries = 30
         for _ in range(0, tries):
-            cmd = "dmesg | grep 'permanently disabled'; echo $?"
-            if int(process.system_output(cmd, ignore_status=True,
-                                         shell=True).decode("utf-8")[-1]) == 0:
+            cmd = "(dmesg | grep 'permanently disabled'; echo $?)"
```
Comment (narasimhan-v) [inline io/pci/EEH.py:247]: While on it, can you change this to python parsing instead of grep please ?

File: io/pci/EEH.py.data/EEH.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+scenario:
```
Comment (narasimhan-v) [inline io/pci/EEH.py.data/EEH.yaml:1]: `scenario` is not needed

**Review by narasimhan-v — CHANGES_REQUESTED** [review]

**Comment by smruti77** [comment]
[EEH.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/6001059/EEH.log)

File: io/pci/EEH.py [hunk 4]
```diff
@@ -220,10 +244,20 @@ def check_eeh_removed(cls):
         """
         tries = 30
         for _ in range(0, tries):
-            cmd = "dmesg | grep 'permanently disabled'; echo $?"
-            if int(process.system_output(cmd, ignore_status=True,
-                                         shell=True).decode("utf-8")[-1]) == 0:
+            cmd = "dmesg; echo $?"
+            res = process.system_output(cmd, ignore_status=True,
+                                        shell=True).decode("utf-8")
+            if 'permanently disabled' in res and int(res[-1]) == 0:
```
Comment (narasimhan-v) [inline io/pci/EEH.py:250]: echo $? is also not needed.
We can compare it using process.exit_status

File: io/pci/EEH.py [hunk 5]
```diff
@@ -132,31 +147,48 @@ def test_eeh_basic_pe(self):
...
-               && lspci; echo $?" % (addr, err, func, mem_addr, mask, phb)
-        return int(process.system_output(cmd, ignore_status=True,
-                                         shell=True).decode("utf-8")[-1])
+        if self.is_baremetal():
+            cmd = "echo %s:%s:%s:%s:%s > /sys/kernel/debug/powerpc/PCI%s/err_injct \
+                   && lspci; echo $?" % (addr, err, func, pci_mem_addr, pci_mask, phb)
+            return int(process.system_output(cmd, ignore_status=True,
+                                             shell=True).decode("utf-8")[-1])
+        else:
+            cmd = "errinjct eeh -v -f %s -s %s/%s -a %s -m %s; echo $?"\
```
Comment (narasimhan-v) [inline io/pci/EEH.py:185]: echo $? is also not needed.
We can compare it using process.exit_status

**Review by narasimhan-v — CHANGES_REQUESTED** [review]
@smruti77 Since we are enhancing this code, can we clean the code to remove most of bash processing please.

**Comment by smruti77** [comment]
[powernveeh.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/6099279/powernveeh.log)
[powervmeeh.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/6099281/powervmeeh.log)

**Review by narasimhan-v — APPROVED** [review]

**Pull request merged**